firmware/powerboard2/code/app/housingMonitor.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * housingMonitor.c
+ *
+ *  Created on: Sep 7, 2020
+ *      Author: Shuhang Tan
+ */
+
+#include "housingMonitor.h"
+#include "ADC.h"
+
+
+#define threshold (2048)
+housingMonitor_housingStatus_E HStatus;
+
+housingMonitor_housingStatus_E housingMonitor_getHousingStatus(void)
+{
+	return HStatus;
+}
+
+// Actual detection of leakage
+void housingMonitor_run(void)
+{
+	HStatus = HOUSINGMONITOR_HOUSINGSTATUS_NO_LEAK;

```

Comment:
Set this in the init function. Once we detect a leak, the housing status to latch to a leak state and not go back to no leak.